src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -35,36 +37,80 @@ trait EntityComponent {
     type EntityQuery = Query[EntityTable, EntityRecord, Seq]
     type EntityQueryWithAttributesAndRefs =  Query[(EntityTable, Rep[Option[(EntityAttributeTable, Rep[Option[EntityTable]])]]), (EntityRecord, Option[(EntityAttributeRecord, Option[EntityRecord])]), Seq]
 
-    implicit val getEntityRecord = GetResult { r => EntityRecord(r.<<, r.<<, r.<<, r.<<, r.<<) }
+    private object EntityRecordRawSqlQuery extends RawSqlQuery {
+      val driver: JdbcDriver = EntityComponent.this.driver
+      implicit val getEntityRecord = GetResult { r => EntityRecord(r.<<, r.<<, r.<<, r.<<, r.<<) }
 
-    // result structure from entity and attribute list raw sql
-    case class EntityListResult(entityRecord: EntityRecord, attributeRecord: Option[EntityAttributeRecord], refEntityRecord: Option[EntityRecord])
+      def action(workspaceId: UUID, entities: Traversable[AttributeEntityReference]) = {
+        val baseSelect = sql"select id, name, entity_type, workspace_id, record_version from ENTITY where workspace_id = $workspaceId and (entity_type, name) in ("
+        val entityTypeNameTuples = reduceSqlActionsWithDelim(entities.map { case entity => sql"(${entity.entityType}, ${entity.entityName})" }.toSeq)
+        concatSqlActions(baseSelect, entityTypeNameTuples, sql")").as[EntityRecord]
+      }
+    }
 
-    // tells slick how to convert a result row from a raw sql query to an instance of EntityListResult
-    implicit val getEntityListResult = GetResult { r =>
-      // note that the number and order of all the r.<< match precisely with the select clause of baseEntityAndAttributeSql
-      val entityRec = EntityRecord(r.<<, r.<<, r.<<, r.<<, r.<<)
+    private object EntityAndAttributesRawSqlQuery extends RawSqlQuery {
```

Comment:
this is a more complicated RawSqlQuery. It has a custom case class for the result and more than 1 action method because there are different where clauses.